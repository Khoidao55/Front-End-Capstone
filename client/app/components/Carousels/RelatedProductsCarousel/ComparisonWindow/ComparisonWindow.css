#comparison-modal-containter {
  position: absolute;
  left: 50%;
  top: 50%;
  z-index: 6;
  width: 400px;
  height: 450px;
  border: 3px solid black;
  background-color: white;
}

.comparison-modal-invisible {
  display: none;
}

#comparison-modal-title {
  text-align: center;
}

#comparison-modal-names {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

#comparison-modal-current-name {
  margin-left: 15px;
}

#comparison-modal-related-name {
  margin-right: 15px;
}