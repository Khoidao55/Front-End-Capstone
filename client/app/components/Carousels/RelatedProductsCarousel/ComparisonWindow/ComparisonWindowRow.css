#comparison-modal-row {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  margin: 10px 0;
}

#comparison-row-current-value{
  text-align: left;
}

#comparison-row-feature{
  text-align: center;
  font-style: italic;
}

#comparison-row-related-value{
  text-align: right;

}