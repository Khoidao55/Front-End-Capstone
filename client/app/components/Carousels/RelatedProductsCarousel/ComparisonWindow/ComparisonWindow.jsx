import React from 'react';
import propTypes from 'proptypes';
import './ComparisonWindow.css';

// features = [ {feature, value}, {feature, value} ]

const ComparisonWindow = ({ displayModal }) => {
  // Hard Coded values for testing
  const currentProductName = 'Camo Onesie';
  const currentProductFeatures = [
    {
      feature: 'Fabric',
      value: 'Canvas',
    },
    {
      feature: 'Buttons',
      value: 'Brass',
    },
  ];
  const relatedProductName = 'Morning Joggers';
  const relatedProductFeatures = [
    {
      feature: 'Fabric',
      value: '100% Cotton',
    },
    {
      feature: 'Cut',
      value: 'Skinny',
    },
  ];
  console.log(currentProductName, '||', relatedProductName);
  return (
    <div
      id="comparison-modal-containter"
      className={
        displayModal
          ? 'comparison-modal-visible'
          : 'comparison-modal-invisible'
      }
    >
      <h3 id="comparison-modal-title">Comparison</h3>
      <div id="comparison-modal-names">
        <div id="comparison-modal-current-name">{currentProductName}</div>
        <div id="comparison-modal-related-name">{relatedProductName}</div>
      </div>
    </div>
  );
};

ComparisonWindow.propTypes = {
  displayModal: propTypes.bool.isRequired,
};

export default ComparisonWindow;
