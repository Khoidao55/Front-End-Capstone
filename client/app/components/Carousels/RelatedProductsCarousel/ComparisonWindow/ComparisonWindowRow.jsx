import React from 'react';
import propTypes from 'proptypes';
import './ComparisonWindowRow.css';

const ComparisonWindowRow = ({
  comparedFeature = 'Compared Feature',
  currentProductValue = '',
  relatedProductValue = '',
}) => (
  <div id="comparison-modal-row">
    <div id="comparison-row-current-value">{currentProductValue}</div>
    <div id="comparison-row-feature">{comparedFeature}</div>
    <div id="comparison-row-related-value">{relatedProductValue}</div>
  </div>
);

ComparisonWindowRow.propTypes = {
  comparedFeature: propTypes.string.isRequired,
  currentProductValue: propTypes.string.isRequired,
  relatedProductValue: propTypes.string.isRequired,
};

export default ComparisonWindowRow;
