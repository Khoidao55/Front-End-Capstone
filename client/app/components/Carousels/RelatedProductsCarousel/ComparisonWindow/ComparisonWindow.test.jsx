import React from 'react';
import { shallow } from 'enzyme';
import ComparisonWindow from './ComparisonWindow.jsx';
import ComparisonWindowRow from './ComparisonWindowRow.jsx';

const dummyModal = {
  displayModal: true,
  currentProductName: 'pretty cool bananas',
  currentProductFeatures: [{ feature: 'bananas', value: 'pretty cool' }, { feature: 'color', value: 'yellow' }],
  relatedProductName: 'donkey kong',
  relatedProductFeatures: [{ feature: 'swirl', value: 'part of his head' }, { feature: 'color', value: 'brown' }],
};

const dummyModalRow = {
  comparedFeature: 'color',
  currentProductValue: 'yellow',
  relatedProductValue: 'brown',
};

describe('ComparisonWindow component', () => {
  test('ComparisonWindow component renders correctly', () => {
    const wrapper = shallow(<ComparisonWindow { ...dummyModal } />);

    expect(wrapper).toMatchSnapshot();
  });

  test('ComparisonWindowRow component renders correctly', () => {
    const wrapper = shallow(<ComparisonWindowRow { ...dummyModalRow } />);

    expect(wrapper).toMatchSnapshot();
  });
});
