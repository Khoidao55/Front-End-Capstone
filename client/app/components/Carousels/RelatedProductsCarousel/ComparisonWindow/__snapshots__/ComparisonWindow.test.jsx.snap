// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonWindow component ComparisonWindow component renders correctly 1`] = `ShallowWrapper {}`;

exports[`ComparisonWindow component ComparisonWindowRow component renders correctly 1`] = `ShallowWrapper {}`;
